### PR TITLE
Decode pattern

### PR DESCRIPTION
**Pull request by biggieman**
I think this is the easiest way to do with backward compatibility
#24 #34

### CODE REVIEW

File: dist/mention.js [hunk 1]
```diff
@@ -145,7 +159,7 @@ angular.module('ui.mention').controller('uiMention', ["$element", "$scope", "$at
   this.decode = function () {
     var value = arguments.length <= 0 || arguments[0] === undefined ? ngModel.$modelValue : arguments[0];
 
-    return value ? value.replace(/@\[([\s\w]+):[0-9a-z-]+\]/gi, '$1') : '';
+    return value ? value.replace(this.decodePattern(), '$1') : '';
```
Comment (ProLoser) [inline dist/mention.js]: Wouldn't it make more sense to make this a property instead of a function, similar to `this.pattern`?

**Review by ProLoser — CHANGES_REQUESTED** [review]

File: dist/mention.js (hunk 1 above)
Comment (biggieman) [reply dist/mention.js]: I chose between function and property. I looked at the function `this.encode` and I think, that `this.decodePattern` is similar to `this.encode` logic.

Comment (ProLoser) [reply dist/mention.js]: But there's literally `this.decode()` to go with `this.encode()` so wouldn't `this.pattern` become `this.encodePattern` and then `this.decodePattern` would match it?

Comment (biggieman) [reply dist/mention.js]: But `this.pattern` is not used by `this.encode()` function. 
I think that the best name for this property is `this.triggerPattern` or `this.searchPattern`.

And I agree that `this.decodePattern` can be a property.

Comment (ProLoser) [reply dist/mention.js]: Touché.

Comment (ProLoser) [reply dist/mention.js]: I pushed a new travis file to master, if you rebase your PR it should fix your build errors.

**Comment by biggieman** [comment]
Changed `this.decodePattern()` from function to property. Renamed `this.pattern` to `this.searchPattern`

**Review by ProLoser — APPROVED** [review]

**Comment by ProLoser** [comment]
Looks good. I may leave this to you to merge in as this is a slightly backwards incompatible change and should probably be pushed to a new (at least minor) version release.

**Comment by biggieman** [comment]
@ProLoser As you wish. And i also could change back to `this.pattern` for backward compatibility.

**Comment by ProLoser** [comment]
I'm indifferent.

**Pull request closed without merging**